lib/http/HttpConnectionsStream.js

```diff
@@ -35,6 +35,7 @@ HttpConnectionsStream.prototype.close = function () {
 
 HttpConnectionsStream.prototype._fetchNextPage = function () {
   var self = this;
+  this.emit('request');
```

Comment:
Yes: you need to make Fetcher.js inherit from EventEmitter. Only then you'll be able to emit events :)